### ISSUE TITLE
Fix Request: SynchronizingForLocalVars false positive

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#204]
A field is properly guarded by a lock:
```
@Data
@AllArgsConstructor
public class Information {
    private int delay;
    private final Object delayLock = new Object();

    public Information() { }

    public void setDelay(int delay) {
        synchronized (delayLock) {
            this.delay = delay;
        }
    }

    public int getDelay() {
        synchronized(delayLock) { // false positive
            return this.delay;
        }
    }
}
```
